index.js

```diff
@@ -1,39 +1,52 @@
-var through2 = require('through2');
+'use strict';
+var Readable = require('stream').Readable;
+var util = require('util');
 
 var micstream = module.exports = function micstream(stream, opts) {
-  var bufferSize = 4096, inputChannels = 2, outputChannels = 2;
+  // "It is recommended for authors to not specify this buffer size and allow the implementation to pick a good
+  // buffer size to balance between latency and audio quality."
+  // https://developer.mozilla.org/en-US/docs/Web/API/AudioContext/createScriptProcessor
+  // Possible values: null, 256, 512, 1024, 2048, 4096, 8192, 16384
+  var bufferSize = null;
   opts = opts || {};
 
   bufferSize = opts.bufferSize || bufferSize;
 
-  if (opts.channels)
-    inputChannels = outputChannels = opts.channels;
+    // We can only emit one channel's worth of audio, so only one input. (Who has multiple microphones anyways?)
+  var inputChannels = 1,
+    // And we're not sending any audio back to the WebAudio API, so don't make it set aside resources for that.
+    outputChannels = 0;
 
-  inputChannels = opts.inputChannels || inputChannels;
-  outputChannels = opts.outputChannels || outputChannels;
-
-  var consumerStream = through2.obj();
+  Readable.call(this, opts);
 
+  var self = this;
+  var recording = true;
   function recorderProcess(e) {
-    var input = e.inputBuffer;
-    var output = e.outputBuffer;
-
-    var channels = [];
-    for (var channel = 0; channel < input.numberOfChannels; channel++) {
-      channels.push(input.getChannelData(channel));
+    if (recording) { // onaudioprocess can be called at least once after we've stopped
+      var raw = e.inputBuffer.getChannelData(0); // Float32Array, each sample is a number from -1 to 1
+      var nodebuffer = new Buffer(raw.buffer); // node only accepts buffers or strings for streams, buffers are essentially a Uint8Array
```

Comment:
Normally I would say absolutely yes, but in this case I don't think so because every browser that [supports getUserMedia/Stream API](http://caniuse.com/#search=getusermedia) also [supports TypedArrays](http://caniuse.com/#search=typedarray)

(It's been kind of nice for this project that I can basically stick to modern standards and not feel bad about it. I wouldn't let my self use [fat arrows](http://caniuse.com/#search=arrow) because even though all of my target browsers support it, it's a parse error in older browsers and I'd like to at least give a clean error message. But that's been about it.)